File: day2/python/part1.py
with open("../input.txt") as file:
    instructions = file.read().splitlines();
    
    
pos = [0,0]
aim = 0
for ins in instructions:
    ins = ins.split(" ")
    direction = ins[0]
    amount = int(ins[1])
    if direction == "forward":
        pos[0] += amount
    if direction == "down":
        pos[1] += amount
    if direction == "up":
        pos[1] -= amount
        
print(pos[0]*pos[1])

File: day2/haskell/part1.hs
module Main where

import Data.Char (isAlpha)

main :: IO ()
main = do
  entries <- map (span isAlpha) <$> (lines <$> readFile "../input.txt")
  let pos = (0, 0)
  print $ uncurry (*) $ foldr increase pos entries

increase :: (String, String) -> (Integer, Integer) -> (Integer, Integer)
increase (direction, n) (x, y) =
  let amount = read n :: Integer
   in case direction of
        "forward" -> (x + amount, y)
        "down" -> (x, y + amount)
        "up" -> (x, y - amount)
        _ -> error "invalid insturction"
